pkg/state/scripts_storage.go

```diff
@@ -69,40 +69,56 @@ func (as *assetScripRecordForHashes) less(other stateComponent) bool {
 	return bytes.Compare(as.asset, as2.asset) == -1
 }
 
-func scriptExists(recordBytes []byte) bool {
-	// Detect if script length is not 0 without unmarshal.
-	return len(recordBytes) > crypto.KeySize
-}
-
 type scriptRecord struct {
-	pk     crypto.PublicKey
-	script proto.Script
+	Data proto.Script `cbor:"0,keyasint,omitemtpy"`
```

Comment:
I think it simpler to store script bytes without wrapping them in CBOR. Do we plan to extend this structure later? Because it looks like this is the only reason to use CBOR encoding here. Giving up CBOR here will save us a little time on encoding/decoding.

Reply:
Consider to remove `scriptRecord` completely.